src/main/java/org/folio/entitlement/configuration/RetryConfiguration.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.entitlement.configuration;
+
+import jakarta.ws.rs.WebApplicationException;
+import java.util.function.Predicate;
+import org.folio.entitlement.integration.IntegrationException;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.backoff.BackOffPolicyBuilder;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+@EnableRetry
+@Configuration
+public class RetryConfiguration {
+
+  @Value("${retries.module.max:3}") private int maxRetriesForFolioModuleCalls = 3;
+  @Value("${retries.keycloak.max:3}") private int maxRetriesForKeycloakCalls = 3;
+
+  @Value("${retries.module.backoff.delay:1000}") private int backOffDelayForFolioModuleCalls = 1000;
+  @Value("${retries.module.backoff.maxdelay:30000}") private int backOffMaxDelayForFolioModuleCalls = 30000;
+  @Value("${retries.module.backoff.multiplier:5}") private int backOffMultiplierForFolioModuleCalls = 5;
+
+  @Value("${retries.keycloak.backoff.delay:1000}") private int backOffDelayForKeycloakCalls = 1000;
+  @Value("${retries.keycloak.backoff.maxdelay:30000}") private int backOffMaxDelayForKeycloakCalls = 30000;
+  @Value("${retries.keycloak.backoff.multiplier:5}") private int backOffMultiplierForKeycloakCalls = 5;
```

Comment:
it can be moved to configuration properties (common approach for providing configuration in Eureka team)